### ISSUE TITLE
[BUG] aux loss will be saved twice when recompute-granularity is full

### ISSUE THREAD

**Issue by bescks** [NVIDIA/Megatron-LM#1514]
If using recompute, the function `save_to_aux_losses_tracker` will be executed twice(both in forward and checkpoint) and the aux_loss will be accumulated twice also. finally, the saved aux loss is two times as much as true aux loss
https://github.com/NVIDIA/Megatron-LM/blob/core_r0.11.0/megatron/core/transformer/moe/router.py#L228
```
    def apply_load_balancing_loss(
        self, activation: torch.Tensor, load_balancing_loss_func: Callable
    ):
        """Calculate auxiliary loss, attach gradient function to activation and add to logging."""
        moe_aux_loss_coeff = self.config.moe_aux_loss_coeff
        if moe_aux_loss_coeff == 0:
            return activation
        sequence_partition_group = None
        if self.config.moe_token_dispatcher_type == "alltoall_seq":
            sequence_partition_group = parallel_state.get_context_parallel_group()
            moe_aux_loss_coeff /= parallel_state.get_tensor_model_parallel_world_size()
        elif parallel_state.get_tensor_and_context_parallel_world_size() > 1:
            sequence_partition_group = parallel_state.get_tensor_and_context_parallel_group()

        aux_loss = load_balancing_loss_func(
            moe_aux_loss_coeff=moe_aux_loss_coeff, sequence_partition_group=sequence_partition_group
        )
        save_to_aux_losses_tracker(
            "load_balancing_loss",
            aux_loss / moe_aux_loss_coeff,
            self.layer_number,
            self.config.num_layers,
            reduce_group=sequence_partition_group,
        )
        activation = MoEAuxLossAutoScaler.apply(activation, aux_loss)
        return activation
```

**Environment (please complete the following information):**
 - Megatron-LM commit ID
 tag core_r0.11.0
 - PyTorch version
 - CUDA version
 - NCCL version

**Comment by sbhavani** [comment]
This should be resolved by e6d56d6828c0773f55772b92b2ec0eed5639665e. Please reopen with a minimal repro if you still see it on latest main.
